### PR TITLE
Add build check and release draft workflows

### PR DESCRIPTION
**Pull request by ADawesomeguy**
There are two workflows added:
1. Check build on any push to make sure it's functional
2. Draft a non-public release on tag

~~<sub>these are totally not copied from the ones I wrote for aeltoth</sub>~~

### CODE REVIEW

File: .github/workflows/draft-release.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Build with Cargo
+      uses: actions-rs/cargo@v1
+      with:
+        use-cross: true
+        command: build
+        args: --release --target=x86_64-pc-windows-gnu
+    - name: Create draft release
+      uses: ncipollo/release-action@v1
+      with:
+        artifacts: "target/x86_64-pc-windows-gnu/release/tw3-modlist-manager.exe"
```
Comment (SpontanCombust) [inline .github/workflows/draft-release.yml:33]: Wrong file name :)

Comment (SpontanCombust) [reply .github/workflows/draft-release.yml:33]: It should be "target/x86_64-pc-windows-gnu/release/rw3d_cli.exe"

Comment (ADawesomeguy) [reply .github/workflows/draft-release.yml:33]: Oh yeah we ignore the copy-pasting 😳

**Comment by SpontanCombust** [comment]
Thank you. 
This will very much come in handy.

**Pull request merged**